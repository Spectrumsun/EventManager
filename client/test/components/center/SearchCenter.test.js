import React from 'react';
import { mount, shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import render from 'react-test-renderer';
import { createMemoryHistory } from 'history';
import thunk from 'redux-thunk';
import sinon from 'sinon';
import ConnectedSearch,
{ Search } from '../../../src/components/Center/SearchCenter';
import Display from '../../../src/components/Center/getCenter'


const middleware = [thunk];
const mockStore = configureStore(middleware);

const initialState = {
  search: {
    searchReuslt: {
      message: 'success',
      match: 'lol',
      searchString: 'lol',
      result: [
        {
          id: 29,
          centerName: 'farm center',
          city: 'Yaba',
          address: 'No 33, yaba road lagos',
          availability: 'yes',
          imageurl: 'llal',
          imageId: 'lol',
          about: 'lalalla',
          facility: ['car', 'fan', 'video']
        },
        {
          id: 9,
          centerName: 'farm center',
          city: 'Yaba',
          address: 'No 33, yaba road lagos',
          availability: 'yes',
          imageurl: 'llal',
          imageId: 'lol',
          about: 'lalalla',
          facility: ['car', 'fan', 'video']
        }
      ],
      count: 10,
      pages: 2
    },
    error: false
  }
};

const store = mockStore(initialState);

const props = {
  onSearch: sinon.spy(() => new Promise((cb) => {
    cb();
  })),
  history: createMemoryHistory(),
  searchResult: {
    message: 'success',
    match: 'lol',
    searchString: 'lol',
    result: [
      {
        id: 29,
        centerName: 'farm center',
        city: 'Yaba',
        address: 'No 33, yaba road lagos',
        availability: 'yes',
        imageurl: 'llal',
        imageId: 'lol',
        about: 'lalalla',
        facility: ['car', 'fan', 'video']
      },
      {
        id: 9,
        centerName: 'farm center',
        city: 'Yaba',
        address: 'No 33, yaba road lagos',
        availability: 'yes',
        imageurl: 'llal',
        imageId: 'lol',
        about: 'lalalla',
        facility: ['car', 'fan', 'video']
      }
    ],
    count: 10,
    pages: 2
  },
  error: false
  
};

const mountedWrapper = mount(
  <Provider store={store}>
    <BrowserRouter>
      <ConnectedSearch {...props} />
    </BrowserRouter>
  </Provider>);

const shallowWrapper = shallow(<Search {...props} />);

const state = {
  totalPage: 5,
  next: 1,
  search: 'tomato',
  pageNmber: 1
};

const event = {
  preventDefault: jest.fn(),
  target: {
    totalPage: 5,
    next: 1,
    search: 'tomato',
    pageNmber: 1
  }
};


// let wrapper;

describe('<Search /> Component', () => {
  const wrapper = shallow(<Search {...props} />);
  it('should render the <Search />', () => {
    shallow(<Search {...props} />);
  });


  it('should match component snapshot', () => {
    const tree = render.create(<Provider store={store}>
      <BrowserRouter>
        <ConnectedSearch {...props} />
      </BrowserRouter>
    </Provider>);
    expect(tree).toMatchSnapshot();
  });


  it('should render initial layout of <Search />', () => {
    expect(wrapper.getElements()).toMatchSnapshot();
  });

  it('should have three div on layout', () => {
    expect(wrapper.find('div').length).toEqual(4);
  });

  it('should have <Display /> when the page loads', () => {
    expect(wrapper.find(Display)).toHaveLength(2);
  });

  it('calls add event when the previous button is clicked for pagination', () => {
    sinon.spy(shallowWrapper.instance(), 'add');
    shallowWrapper.setState(state);
    shallowWrapper.instance().add();
    expect(shallowWrapper.instance().add.calledOnce).toEqual(true);
  });

  it('calls minus event when the next button is clicked for pagination', () => {
    sinon.spy(shallowWrapper.instance(), 'minus');
    shallowWrapper.setState(state);
    shallowWrapper.instance().minus();
    expect(shallowWrapper.instance().minus.calledOnce).toEqual(true);
  });
});

