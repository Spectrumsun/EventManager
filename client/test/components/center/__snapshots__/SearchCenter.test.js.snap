// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> Component should match component snapshot 1`] = `
<div>
  <div
    className="center "
    style={
      Object {
        "paddingTop": "100px",
      }
    }
  >
    <h1
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      lol
    </h1>
    <div>
      <a
        href="/centers/29"
        onClick={[Function]}
        style={
          Object {
            "color": "black",
          }
        }
      >
        <div
          className="card d-lg-inline-block"
          onClick={undefined}
          style={
            Object {
              "width": "14rem",
            }
          }
        >
          <img
            alt="Card cap"
            className="card-img-top"
            src="llal"
            style={
              Object {
                "height": "15rem",
                "objectFit": "cover",
              }
            }
          />
          <div
            className="card-body"
          >
            <h4
              className="card-title"
            >
              farm
            </h4>
            <h6
              className="card-subtitle mb-2 text-muted"
            >
              Yaba
            </h6>
            <h5
              className="card-link"
              id="centerDetails"
            >
              View More
            </h5>
          </div>
        </div>
      </a>
      <a
        href="/centers/9"
        onClick={[Function]}
        style={
          Object {
            "color": "black",
          }
        }
      >
        <div
          className="card d-lg-inline-block"
          onClick={undefined}
          style={
            Object {
              "width": "14rem",
            }
          }
        >
          <img
            alt="Card cap"
            className="card-img-top"
            src="llal"
            style={
              Object {
                "height": "15rem",
                "objectFit": "cover",
              }
            }
          />
          <div
            className="card-body"
          >
            <h4
              className="card-title"
            >
              farm
            </h4>
            <h6
              className="card-subtitle mb-2 text-muted"
            >
              Yaba
            </h6>
            <h5
              className="card-link"
              id="centerDetails"
            >
              View More
            </h5>
          </div>
        </div>
      </a>
    </div>
    <ul
      className="pagination nav justify-content-center"
    >
      <li
        className="page-item"
        onClick={[Function]}
      >
        <a
          className="page-link"
        >
          Previous
        </a>
      </li>
      <li
        className="page-item"
      >
        <a
          className="page-link"
        >
          Page 1  of 
          2
        </a>
      </li>
      <li
        className="page-item"
        onClick={[Function]}
      >
        <a
          className="page-link"
        >
          Next
        </a>
      </li>
    </ul>
    <div
      style={
        Object {
          "marginBottom": "100px",
        }
      }
    />
  </div>
  <div
    className="center"
  >
    <footer>
      <div>
        <h3
          className="center"
        >
          Event Manager
        </h3>
      </div>
      <div
        style={
          Object {
            "display": "inline-block",
            "textAlign": "left",
          }
        }
      >
        <div>
          <div
            className="float-left"
          >
            <a
              className="navbar-brand"
              href="/"
            >
              <img
                alt=""
                height="30"
                src="https://res.cloudinary.com/skybound/image/upload/v1522444986/eventmanager/static/event.png"
                width="30"
              />
            </a>
          </div>
          <div
            className="float-left"
          >
            <ul>
              <p
                className="footerTitle"
              >
                Social Media
              </p>
              <li
                className="footerColor"
              >
                facebook
              </li>
              <li
                className="footerColor"
              >
                twitter
              </li>
              <li
                className="footerColor"
              >
                instargram
              </li>
            </ul>
          </div>
          <div
            className="float-left"
          >
            <ul>
              <p
                className="footerTitle"
              >
                Event manager
              </p>
              <li
                className="footerColor"
              >
                Questions
              </li>
              <li
                className="footerColor"
              >
                Jobs
              </li>
              <li
                className="footerColor"
              >
                Help
              </li>
            </ul>
          </div>
          <div
            className="float-left"
          >
            <ul>
              <p
                className="footerTitle"
              >
                About
              </p>
              <li
                className="footerColor"
              >
                desgin by Taiwo Sunday
              </li>
              <li
                className="footerColor"
              >
                +2348038512455
              </li>
              <li
                className="footerColor"
              >
                https://github.com/Spectrumsun
              </li>
              <li
                className="footerColor"
              >
                taiwo.sunday@outlook.com
              </li>
            </ul>
          </div>
          <div
            className="float-left"
          >
            <ul>
              <p
                className="footerTitle"
              >
                Site
              </p>
              <li
                className="footerColor"
              >
                Home
              </li>
              <li
                className="footerColor"
              >
                Centers
              </li>
              <li
                className="footerColor"
              >
                Events
              </li>
            </ul>
          </div>
        </div>
      </div>
      <p
        className="center"
      >
        Event Managment Â© 2017
      </p>
    </footer>
  </div>
</div>
`;

exports[`<Search /> Component should render initial layout of <Search /> 1`] = `
Array [
  <div>
    <div
      className="center "
      style={
        Object {
          "paddingTop": "100px",
        }
      }
    >
      <h1
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        lol
      </h1>
      <div>
        <Link
          replace={false}
          style={
            Object {
              "color": "black",
            }
          }
          to="/centers/29"
        >
          <getCenter
            address="No 33, yaba road lagos"
            centerName="farm"
            city="Yaba"
            image="llal"
          />
        </Link>
        <Link
          replace={false}
          style={
            Object {
              "color": "black",
            }
          }
          to="/centers/9"
        >
          <getCenter
            address="No 33, yaba road lagos"
            centerName="farm"
            city="Yaba"
            image="llal"
          />
        </Link>
      </div>
      <ul
        className="pagination nav justify-content-center"
      >
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
          >
            Previous
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
          >
            Page 1  of 
            2
          </a>
        </li>
        <li
          className="page-item"
          onClick={[Function]}
        >
          <a
            className="page-link"
          >
            Next
          </a>
        </li>
      </ul>
      <div
        style={
          Object {
            "marginBottom": "100px",
          }
        }
      />
    </div>
    <footer />
  </div>,
]
`;
